test/controllers/users_controller_test.rb

```diff
@@ -2,49 +2,180 @@
+        User.role.values.reject { |role| role == :admin }.each do |role|
+          @user.update role: role
+          patch send("change_role_to_#{endpoint}_path".to_sym, @user_2)
+          assert_response :redirect
+        end
+      end
+    end
+
+    it 'should respond unauthorized if user is not admin - users_search' do
       User.role.values.reject { |role| role == :admin }.each do |role|
         @user.update role: role
-        assert_no_difference 'User.count' do
-          post users_path, params: { user: attributes_for(:user) }
-        end
+        post users_search_path, params: { search: '' }, xhr: true
         assert_response :unauthorized
       end
     end
-  end
 
-  describe 'index method' do
-    it 'should redirect if user is not admin' do
+    it 'should redirect if user is not admin - user update' do
       User.role.values.reject { |role| role == :admin }.each do |role|
         @user.update role: role
-        get users_path
+        patch user_path(@user_2), params: { user: {} }
         assert_response :redirect
       end
     end
 
-    it 'should let admin access the route' do
-      @user.role = :admin
-      @user.save!
+    it 'should respond unauthorized if user is not admin - user update' do
+      User.role.values.reject { |role| role == :admin }.each do |role|
+        @user.update role: role
+        assert_no_difference 'User.count' do
+          post users_path, params: { user: attributes_for(:user) }
+        end
+        assert_response :unauthorized
+      end
+    end
+  end
+
+  describe 'index method' do
+    it 'should let admins access the route' do
       get users_path
       assert_response :success
     end
   end
 
+  describe 'edit method' do
+    it 'should be accessible' do
+      get edit_user_path(@user)
+      assert_response :success
+    end
+  end
+
   describe 'search method' do
+    it 'should work on empty search' do
+      post users_search_path, params: { search: '' }, xhr: true
+      assert_response :success
+    end
+
+    it 'should work on nonempty search' do
+      post users_search_path, params: { search: 'yolo' }, xhr: true
+      assert_response :success
+    end
+  end
+
+  describe 'new method' do
     it 'should return success' do
-      post users_search_path, params: { search: '' }, xhr: :true
+      get new_user_path
       assert_response :success
     end
   end
 
+  describe 'create method' do
+    it 'should create a user' do
+      attributes = attributes_for(:user)
+      attributes.delete :role
+      assert_difference 'User.count', 1 do
+        post users_path, params: { user: attributes }
+      end
+    end
+
+    it 'should show errors if creation fails' do
+      attributes = attributes_for(:user)
+      attributes[:name] = nil
+      assert_no_difference 'User.count' do
+        post users_path, params: { user: attributes }
+      end
+      assert_includes response.body, 'can&#39;t be blank'
```

Comment:
What an odd assertion

Reply:
haha. it escapes the apostrophe like a real jerk.